### PR TITLE
Fixed Universe chainID quick selector not grouping segments

### LINKED ISSUE #1107: MMTF parser doesn't work well.
### Expected behaviour
show all residues in A chain:

### Actual behaviour
['HOH', 'HOH', 'HOH']

### Code to reproduce the behaviour

``` python
from MDAnalysis.coordinates.MMTF import fetch_mmtf
s=fetch_mmtf("1kq2")
s.A.residues
list(r.resname for r in s.A.residues)

```

### Currently version of MDAnalysis:
0.16.0-dev0

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #1107

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/core/universe.py [hunk 1]
```diff
@@ -268,13 +268,20 @@ def _generate_from_topology(self):
...
 
         # Update Universe namespace with segids
-        for seg in self.segments:
-            if hasattr(seg, 'segid') and seg.segid:
-                if seg.segid[0].isdigit():
-                    name = 's' + seg.segid
+        # Many segments can have same segid, so group together first
+        try:
+            # returns dict of segid:segment
+            segids = self.segments.groupby('segids')
```
Comment (kain88-de) [inline package/MDAnalysis/core/universe.py]: Good use of the new groubby

File: package/MDAnalysis/core/universe.py [hunk 2]
```diff
@@ -268,13 +268,20 @@ def _generate_from_topology(self):
...
+        except AttributeError:
+            # no segids, don't do this step
+            pass
+        else:
+            for segid, segment in segids.items():
```
Comment (kain88-de) [inline package/MDAnalysis/core/universe.py]: is this supposed to run if the groupby works? I'm not overly familiar with the try/else in python

Comment (richardjgowers) [inline package/MDAnalysis/core/universe.py]: Yes. It's good practice to try and make the try: portion as small as possible so I don't catch and silence an attribute error after the groupby call

**Comment by kain88-de** [comment]
Looks good besides the CI failures

**Comment by kain88-de** [comment]
@richardjgowers this still have test failures that aren't related to the failed test in encore.

File: package/MDAnalysis/core/universe.py [hunk 3]
```diff
@@ -268,13 +290,28 @@ def _generate_from_topology(self):
...
+                    # prefix 's' if starts with number
+                    name = 's' + segid
                 else:
-                    name = seg.segid
-                self.__dict__[name] = seg
+                    name = segid
+                # if len 1 SegmentGroup, convert to Segment
+                if len(segment) == 1:
+                    segment = segment[0]
+                self.__dict__[name] = segment
```
Comment (jbarnoud) [inline package/MDAnalysis/core/universe.py]: I know this is legacy behavior, but what happens if a segment has a name that corresponds already to an attribute of `Universe`?

Comment (richardjgowers) [inline package/MDAnalysis/core/universe.py]: Yeah pretty much what you'd expect. That attribute would stop working. Unless it's a property you can't set, in which case you'd get some weird AttributeError

**Comment by richardjgowers** [comment]
This got `FAIL: test_dres_convergence (MDAnalysisTests.analysis.test_encore.TestEncore)`, but I'm dubious if I managed to break that here, so I'm rerunning tests..

**Comment by jbarnoud** [comment]
Rerunning the tests made the `test_dres_convergence` issue disappear. We should keep in mind that this test failed at random, if it happens again there might be a problem there.

**Comment by orbeckst** [comment]
@kain88-de --- want to merge this or are there any holdups?

**Pull request merged**